app/src/main/java/space/taran/arkmemo/ui/fragments/EditTextNotes.kt

```diff
@@ -0,0 +1,106 @@
+package space.taran.arkmemo.ui.fragments
+
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.View
+import android.widget.EditText
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import by.kirich1409.viewbindingdelegate.viewBinding
+import com.google.android.material.floatingactionbutton.ExtendedFloatingActionButton
+import dagger.hilt.android.AndroidEntryPoint
+import space.taran.arkmemo.R
+import space.taran.arkmemo.data.viewmodels.TextNotesViewModel
+import space.taran.arkmemo.databinding.FragmentEditTextNotesBinding
+import space.taran.arkmemo.models.TextNote
+import space.taran.arkmemo.ui.activities.hideSettingsButton
+
+@AndroidEntryPoint
+class EditTextNotes(): Fragment(R.layout.fragment_edit_text_notes) {
+    constructor(note: TextNote): this(){
+        this.note = note
+    }
+
+    constructor(note: String): this(){
+        noteStr = note
+    }
+
+    private val activity: AppCompatActivity by lazy{
+        requireActivity() as AppCompatActivity
+    }
+
+    private val textNotesViewModel: TextNotesViewModel by activityViewModels()
+
```

Comment:
Why `by activityViewModels()` and not just `by viewModels()`?
The viewmodel in this case follows lifecycle of activity, not the fragment, which is wrong.
Same for other fragments.

Reply:
It is because am using the same ```ViewModel``` for all fragments, I hope they will have to share some data, according to [(https://developer.android.com/topic/libraries/architecture/viewmodel?gclid=CjwKCAjw7p6aBhBiEiwA83fGutYYOPYpe4iac6LBjwaY4bYb5D2aIYZO8mSkuXpQlQUl06hcilHOLhoCVeUQAvD_BwE&gclsrc=aw.ds#kotlin)]

Reply:
I think it's better to use separate viewmodels, since there can be a lot of code and it's difficult to support several screens in one viewmodel. There are other ways to share data.

Reply:
Alright, that makes sense. I will get on it